File: bank-service/src/main/java/rs/edu/raf/mapper/ExchangeRateMapper.java
package rs.edu.raf.mapper;


import org.springframework.stereotype.Component;
import rs.edu.raf.dto.ExchangeRateDto;
import rs.edu.raf.model.ExchangeRate;

@Component
public class ExchangeRateMapper {

    public ExchangeRateDto exchangeRateToExchangeRateResponseDto(ExchangeRate exchangeRate){
        return new ExchangeRateDto(exchangeRate.getCurrencyCode(), exchangeRate.getRate());
    }
}

File: bank-service-kotlin/src/main/kotlin/org/example/bankservicekotlin/mapper/ExchangeRateMapper.kt
package org.example.bankservicekotlin.mapper

import org.example.bankservicekotlin.dto.ExchangeRateDTO
import org.example.bankservicekotlin.model.ExchangeRate
import org.springframework.stereotype.Component

@Component
class ExchangeRateMapper {
    fun exchangeRateToExchangeRateResponseDto(exchangeRate: ExchangeRate): ExchangeRateDTO {
        return ExchangeRateDTO(exchangeRate.currencyCode, exchangeRate.rate)
    }
}